### ISSUE TITLE
v16-19x19_000665-diligent_converted_quantized vs LZ #202 in test match?

### ISSUE THREAD

**Issue by ainixingo** [gcp/leela-zero#2183]
v16-19x19_000665-diligent_converted_quantized.gz

https://drive.google.com/open?id=1Cdfyt7tzlBpWKx2D0zToS-qVgD4goyvl

**Comment by l1t1** [comment]
you can test it with v15 990 first

**Comment by Splee99** [comment]
Just downloaded #665. It seems that still more training is needed for dealing with ladders.

**Comment by l1t1** [comment]
699 and 715 are good

**Comment by Friday9i** [comment]
@l1t1 did you test them, converted or directly with MiniGo?

**Comment by l1t1** [comment]
no, i only saw their elo are very close on https://cloudygo.com/
btw i dont know how to download weights except get from https://cloudygo.com/converted_model/v16-19x19/models/

**Comment by WhiteHalmos** [comment]
Perhaps @sethtroisi can instruct us on where to download weights on cloudygo.com for the v16 run.

**Comment by sethtroisi** [comment]
https://cloudygo.com/converted_model/v16-19x19/models/

**Comment by sethtroisi** [comment]
Also Andrew is trying to play a big series of 990 vs 201(202?) with longer playouts.

**Comment by WhiteHalmos** [comment]
I see only 10 converted models there, are others going to be generated or would convert_minigo.py work on the 40 block models too?

**Comment by WhiteHalmos** [comment]
Also, sorry if this is a dumb question, but I can't seem to find where to download the non-converted models of the v16 run on cloudygo.com either. For example, where to download model 715: https://cloudygo.com/v16-19x19/eval-model/715?sorted=False

**Comment by l1t1** [comment]
I saw some info about weights at https://github.com/tensorflow/minigo#basics,
but I have no access to google, so it's useless for me

**Comment by roy7** [comment]
I don't have the link handy for models, it is in Google someplace. @amj can maybe link it.

But yes you can run the converter script on any of the models.

**Comment by alreadydone** [comment]
@WhiteHalmos Yes you can convert 40b models at https://console.cloud.google.com/storage/browser/minigo-pub/v16-19x19/models using convert_minigo.py locally (if you have Python with Tensorflow installed) or online (https://colab.research.google.com/drive/16eXKg_WswzadfpfbIZzATcSy0qL7F0gC by @godmoves).

**Comment by sethtroisi** [comment]
@WhiteHalmos I don't have links on the site. models are synced to the public bucket immediately so some of the links would be broken.

**Comment by WhiteHalmos** [comment]
@alreadydone thanks for the links!

This is what I ended up running to get converted models in case anyone wanted local commands:
```sh
cd build/run
pipenv shell --python 2
pipenv install gsutil
gsutil config
gsutil ls gs://minigo-pub/v16-19x19/models/000715* | gsutil cp -I .
mv 000715* ../convert
exit
cd ../convert
pipenv shell --python 3.6
pipenv install numpy tensorflow
./convert_minigo.py 000715-trusty
gunzip 000715-trusty_converted.txt.gz
mv 000715-trusty_converted.txt ../run/v16m715
```
Note that python 3.6 is required by tensorflow, and python 2.7 is required by gsutil. I symlinked training/minigo/convert_minigo.py to build/convert and renamed the resulting weights file for easy reference in gomill ringmaster.

**Comment by WhiteHalmos** [comment]
@sethtroisi I assume that there is a delay in uploading to minigo-pub since model 751 is not yet available but cloudygo shows latest model is 795?

```
gsutil ls gs://minigo-pub/v16-19x19/models/000751*
CommandException: One or more URLs matched no objects.``

**Comment by sethtroisi** [comment]
@WhiteHalmos yes there is a delay (upload is manual at this time) but I just resynced them for you

**Comment by WhiteHalmos** [comment]
Thank you! Appreciate it, will fetch the latest ones. :)

**Comment by atoutw** [comment]
@alreadydone 
Thank you for information!
I converted v16-797, and I will try to report some test results later.

**Comment by atoutw** [comment]
![image](https://user-images.githubusercontent.com/43053753/52165960-b89ef880-274a-11e9-84b9-cba0f706e99f.png)
(B)v16-797 (W)LZ202, 1080ti 10s/move

![image](https://user-images.githubusercontent.com/43053753/52165971-e7b56a00-274a-11e9-9ca9-9b599bbf138f.png)
(B)v16-797 (W)LZ202, 2080ti 120s/move

**Comment by sethtroisi** [comment]
So glad we aren't the one being laddered :)

**Comment by WhiteHalmos** [comment]
<details>
<summary>Here are some very preliminary results:</summary>

```
leelaz_lz202_-v10 v leelaz_v16m797_-v10 (103/400 games)                                                                      
board size: 19   komi: 7.5                                                                                                   
                      wins              black         white       avg cpu
leelaz_lz202_-v10       92 89.32%       47 90.38%     45 88.24%     11.81
leelaz_v16m797_-v10     11 10.68%       6  11.76%     5   9.62%     11.37
                                        53 51.46%     50 48.54%
                                                                                                                             
leelaz_lz202_-v100 v leelaz_v16m797_-v100 (103/400 games)                                                                    
board size: 19   komi: 7.5                                                                                                   
                       wins              black         white       avg cpu                                                   
leelaz_lz202_-v100       78 75.73%       43 82.69%     35 68.63%     41.11
leelaz_v16m797_-v100     25 24.27%       16 31.37%     9  17.31%     41.24
                                         59 57.28%     44 42.72%
                                                                                                                             
leelaz_v16m715_-v10 v leelaz_v16m797_-v10 (103/400 games)                                                                    
board size: 19   komi: 7.5                                                                                                   
                      wins              black         white       avg cpu
leelaz_v16m715_-v10     46 44.66%       25 48.08%     21 41.18%     11.31
leelaz_v16m797_-v10     57 55.34%       30 58.82%     27 51.92%     11.42
                                        55 53.40%     48 46.60%
                                                                                                                             
leelaz_v16m715_-v100 v leelaz_v16m797_-v100 (103/400 games)
board size: 19   komi: 7.5
                      wins              black         white       avg cpu                                                    
leelaz_v16m715_-v10     46 44.66%       25 48.08%     21 41.18%     11.31                                                    
leelaz_v16m797_-v10     57 55.34%       30 58.82%     27 51.92%     11.42                                                    
                                        55 53.40%     48 46.60%                                                              

leelaz_v16m715_-v100 v leelaz_v16m797_-v100 (103/400 games)
board size: 19   komi: 7.5
                       wins              black         white       avg cpu                                                   
leelaz_v16m715_-v100     42 40.78%       14 26.92%     28 54.90%     45.54                                                   
leelaz_v16m797_-v100     61 59.22%       23 45.10%     38 73.08%     42.59                                                   
                                         37 35.92%     66 64.08%                                                             

leelaz_lz202_-v1000 v leelaz_v16m797_-v1000 (18/40 games)
board size: 19   komi: 7.5
                        wins              black        white      avg cpu                                                    
leelaz_lz202_-v1000       14 77.78%       7 77.78%     7 77.78%    261.75                                                    
leelaz_v16m797_-v1000      4 22.22%       2 22.22%     2 22.22%    279.76                                                    
                                          9 50.00%     9 50.00%
```
</details>

<details>
<summary>Gomill config</summary>

```
competition_type = 'playoff'
                           
LZ_ARGS = '/home/wnwen/other/leela-zero/build/run/{} --noponder -r10 -g -t1 -d -w {} {}'
def LZ(vals):             
  # time_settings main byoyomi stones
  commands = ['time_settings 0 5 1']
  return Player(LZ_ARGS.format(*vals), startup_gtp_commands=commands)
       
players = dict()
def ADD(p):
  players[p] = LZ(p.split('_'))
       
p1 = 'leelaz_lz202_-v10'
p2 = 'leelaz_v16m797_-v10'
p3 = 'leelaz_lz202_-v100'
p4 = 'leelaz_v16m797_-v100'
p5 = 'leelaz_v16m715_-v10'
p6 = 'leelaz_v16m797_-v10'
p7 = 'leelaz_v16m715_-v100'
p8 = 'leelaz_v16m797_-v100'                              
p9 = 'leelaz_lz202_-v1000'                               
p10 = 'leelaz_v16m797_-v1000'
ADD(p1)
ADD(p2)
ADD(p3)
ADD(p4)
ADD(p5)
ADD(p6)
ADD(p7)
ADD(p8)
ADD(p9)
ADD(p10)

board_size = 19
komi = 7.5
matchups = [
  Matchup(p1, p2, alternating=True, number_of_games=400),
  Matchup(p3, p4, alternating=True, number_of_games=400),
  Matchup(p5, p6, alternating=True, number_of_games=400),
  Matchup(p7, p8, alternating=True, number_of_games=400),
  Matchup(p9, p10, alternating=True, number_of_games=40),
]
```
</detail>

**Comment by l1t1** [comment]
so minigo need more playouts to find ladder?

**Comment by WhiteHalmos** [comment]
Not enough games for -v1000 to know for sure. I'd like to see what -v10000 looks like too but it's a little too slow for my mini 1050ti. 😄

**Comment by l1t1** [comment]
suggest to add more strong 6xx weights to cloudygo converted storage https://cloudygo.com/converted_model/v16-19x19/models/

**Comment by atoutw** [comment]
![image](https://user-images.githubusercontent.com/43053753/52171785-5bd12b80-27a6-11e9-8e10-1cbd1fb017a3.png)
↓
![image](https://user-images.githubusercontent.com/43053753/52171791-7905fa00-27a6-11e9-87f8-b285b17fb17d.png)
v16-805 just took a few moment from 3-3 to 4-4.

![image](https://user-images.githubusercontent.com/43053753/52171818-c84c2a80-27a6-11e9-9c2c-c32fd33b0c2d.png)

![image](https://user-images.githubusercontent.com/43053753/52171854-57f1d900-27a7-11e9-93c9-3e69ec5569c3.png)
But for understanding this ladder (that 797 missed with 120s/move), it took very long time.

**Comment by l1t1** [comment]
you can copy the content of eval games to a sgf and ananlyze it
such as https://cloudygo.com/v16-19x19/805/eval/2019-02-03/1549161182-minigo-cc-evaluator-830-805-wb-4qlsd-0-000805-pallas-000830-merlin.sgf?type=eval
see https://github.com/tensorflow/minigo/issues/678

**Comment by WhiteHalmos** [comment]
@sethtroisi If you get a chance would you mind updating minigo-pub/v16-19x19 with the latest models? Thank you!

**Comment by l1t1** [comment]
8xx seems much better
Ratings from	89536 games	rating	uncertainty
Best model	000882-glory	4452	± 40
2	000894-affray	4436	± 70
3	000805-pallas	4428	± 26
4	000833-black-prince	4419	± 29
5	000883-acheron	4380	± 37
6	000892-alliance	4362	± 45
7	000797-centaur	4360	± 24
8	000866-brave	4353	± 30
9	000856-wanderer	4338	± 30
10	000884-perseus	4328	± 36
11	000864-parthian	4303	± 28
12	000715-trusty	4300	± 20
13	000827-expedition	4262	± 29
14	000699-lively	4256	± 20
15	000863-gibraltar	4256	± 30
Worst model	000002-medusa	0	± 140 Filter bad models    Filter old model

**Comment by ainixingo** [comment]
https://cloudygo.com/converted_model/v16-19x19/models/

**Comment by ainixingo** [comment]
v16-19x19/models/000930-goliath_converted.txt.gz should test with LZ. @gcp

**Comment by l1t1** [comment]
930 is still near best

**Comment by atoutw** [comment]
> 
> 
> @WhiteHalmos Yes you can convert 40b models at https://console.cloud.google.com/storage/browser/minigo-pub/v16-19x19/models using convert_minigo.py locally (if you have Python with Tensorflow installed) or online (https://colab.research.google.com/drive/16eXKg_WswzadfpfbIZzATcSy0qL7F0gC by @godmoves).

@alreadydone 
https://console.cloud.google.com/storage/browser/minigo-pub/v17-19x19/models?pli=1&prefix=00099
I found MiniGo-v17's files are able to download, but seems the same way to convert is not work. Could you please to teach me how to convert them?

**Comment by sethtroisi** [comment]
V17 used squeeze and excitation which is not compatible with LZ

**Comment by SHKD13** [comment]
**@sethtroisi** 
Нello, may I ask how to download a whole folders from Minigo-pub storage at Google Cloud Platform not by clicking every file inside? I mean this link: https://console.cloud.google.com/storage/browser/minigo-pub/?pli=1 
I'd like to download evaluation games starting from v14, if it's possible.

**Comment by sethtroisi** [comment]
@SHKD13 the easiest way is to use [gsutil](https://cloud.google.com/storage/docs/gsutil) which I think you can get via the [ubuntu repo](https://cloud.google.com/storage/docs/gsutil)

You run something like `gsutil -m rsync -r gs://minigo-pub/v14-19x19/eval v14-eval/ v14`

Alternatively I've zipped up v10 to v15 and added them to https://console.cloud.google.com/storage/browser/minigo-pub/eval-games/?pli=1 but those are fairly large file. Let me know if I can make them available in some more useful way.

**Comment by SHKD13** [comment]
@sethtroisi
Thanks a lot, I'll try this util!
About your question: the tar.gz archives size is absolutely not a problem! :) I can see there MiniGo v10, v11, v12 & v13. I've already downloaded them. Would be super great, if it is possible to add there zipped v14, v15, v16 and v17 evaluation games, as well as MiniGo v15 series of games vs Leela Zero #201 and ELF v1, or test games between versions (like v15 Vs. v16 Vs. v17...). MiniGo becoming fairly popular and I'm not the only one who is interesting in its games studying :) 100% sure that we all who likes MiniGo will be extremely grateful for this))

Also me and Sinan Djepov 5d would be happy to take a small advice about how to use **sgfstrip** util (https://homepages.cwi.nl/~aeb/go/sgfutils/html/sgfstrip.html) to clean the debug information out of MiniGo's heavy sgf's. Smaller sgf size will make them processable for using in a big game collections and databases .db files. I have Bash Ubuntu shell on my Windows 10, so technically should not be any difficulties. But it is only if you have a time for such an advices, of course! :)

**Comment by sethtroisi** [comment]
@SHKD13 Sorry the upload of v14,v16, and v16 failed yesterday. They are now uploaded.

I don't quite understand your second paragraph "happy to take a small advice about how to use sgfstrip", but I've created clean versions of the archives before. Can you file an issue on the minigo tracker and I'll can follow up on that next week.

I'll also follow up on a jumble of all the "other" games we've played.

**Comment by SHKD13** [comment]
@sethtroisi 
Thank you very much for these games!
And sorry my English :) I'll try to explain by more simple way. Every MiniGo sgf file contains debug info in a comments for each move, all these numbers and percents which show what MiniGo is thinking. It makes the size of such sgf's in average 130 - 150 kb, while standart sgf size is about 1 - 3 kb. Big size makes MiniGo sgf's unprocessable in a mass for using in databases like Drago or online game collections like Sinan Djepov's AI fuseki tool. 
There is special util to strip this debug info in comments - **sgfstrip** (https://homepages.cwi.nl/~aeb/go/sgfutils/html/sgfstrip.html). Unfortunately, I can't run it by myself only, it's a bit complicated for non programmer man :) So, my only question is it possible to ask some advice from you how to run this util? It is for C++ script for Linux, but I have Bash Ubuntu shell on Windows 10, so it looks possible to run sgfstrip on my laptop. Many thanks from me and any thing I can help you in a future :)

**Comment by sethtroisi** [comment]
@SHKD13 Thanks for clarifying (I wasn't sure if you wanted me to run it, you wanted to run it, or something else). 

I think running the command involves compiling the tool which is a little beyond what I can share here. Would it work if I just run it over all the files for you?

I'd need to do some prework to run to make sure the directories layout stays the same but I can run sgfstrip and produce much smaller archives of each version in the next couple of days if that would be nicer.

**Comment by SHKD13** [comment]
@sethtroisi 
My idea was to save your time and do all the work by myself. But I get the point that not everything in such a tools is sharable :) So I appreciate a lot Your great help with these sgf's, as well as Sinan Djepov and other fans of MiniGo!
To reduce the amount work with archives run sgfstrip starting just from v12 up to v17 versions, maybe also v15 vs LZ & ELF series, or cross-versions games (like v15 Vs. v16) - if such sgf's exist. Don't spend your time for versions earlier v12.
Thanks to You my work with building the Encyclopedia of Go Openings (reimplementation of a Chess ECO code for Go) will be done on a much stronger games ground :)

**Comment by sethtroisi** [comment]
@SHKD13 I uploaded v12 and v15 clean. v13, v14, v16 tomorrow.

**Comment by SHKD13** [comment]
@sethtroisi 
Thank You very much for great help! I've downloaded v12, v13 and v15 :)

**Comment by sethtroisi** [comment]
@SHKD13 I pushed v14 and v16.

I'll work on packaging up MG vs ELF, LZ ... games tomorrow

**Comment by SHKD13** [comment]
@sethtroisi 
Super :) All these games are real treasure for building the modern fuseki classification!

**Comment by SHKD13** [comment]
@sethtroisi
Hi, sorry for asking probably too much with those cross-algorithm games (MG vs LZ & ELF)! :)
May I ask, what is the directory where It is possible to download them without zipping, or maybe these 63117 cross-run-eval games: https://www.cloudygo.com/all-eval-graphs?bucket=v13-19x19
I can try to clean'em by sgfstrip by myself :)

**Comment by sethtroisi** [comment]
@SHKD13 there's a new file ["cross_clean.tar.gz"](https://console.cloud.google.com/storage/browser/minigo-pub/eval-games/?pli=1) that contains many (all? most?) of the cross eval games and random other things we've tried.

**Comment by SHKD13** [comment]
@sethtroisi 
Thank You very much! This is true brilliant games collection for study :)

**Comment by sethtroisi** [comment]
@SHKD13 I added v17 eval games.

**Comment by SHKD13** [comment]
@sethtroisi 
Wow, thanks for a great update! All MiniGo models You've cleaned and place in storage are super, but v17 is my favorite among others))
